org.adempiere.base.process/src/org/compiere/process/RequestEMailProcessor.java

```diff
@@ -457,14 +463,24 @@ protected void createRequest(EmailContent emailContent, String trxName) throws E
 		
 		if (log.isLoggable(Level.INFO)) log.info("created request " + req.getR_Request_ID() + " from msg -> " + emailContent.subject);
 		
-		String htmlContent = emailContent.getHtmlContent(true);
-		if (htmlContent != null){
-			MAttachment attach = req.createAttachment();
-			
-			attach.addEntry(emailContent.subject + ".html", emailContent.getHtmlContent(true).getBytes(Charset.forName("UTF-8")));
-			attach.saveEx(trxName);
+		if("H".equals(p_HTMLAttachmentType)) {
+			String htmlContent = emailContent.getHtmlContent(true);
+			if (htmlContent != null){
+				MAttachment attach = req.createAttachment();
+				
+				attach.addEntry(emailContent.subject + ".html", emailContent.getHtmlContent(true).getBytes(Charset.forName("UTF-8")));
+				attach.saveEx(trxName);
+			}
+		} else if("I".equals(p_HTMLAttachmentType)) {
+			ArrayList<BodyPart> imagesList = emailContent.getHTMLImageBodyParts();
+			for(BodyPart image: imagesList) {

```

Comment:
Looking at the code, a null check for imagesList is necessary here.

Reply:
Commited